episodes/filters-and-thresholding.md

```diff
@@ -0,0 +1,743 @@
+
+## Filters
+
+Let's open the cells image in Napari again:  
+`File > Open Sample > napari builtins > Cells (3D + 2Ch)`
+
+With the new plugins installed, you should see many new options under `Tools` in 
+the top menu-bar of Napari. You can find out more about these options using the 
+plugin's documentation which was linked in the [previous section
+](#adding-plugins-to-help-with-segmentation).
+
+For now, we are interested in the options under:  
+`Tools > Filtering / noise removal`
+
+Specifically, the two options labelled 'Gaussian' - you should see one ending in 
+'n-SimpleITK' and another ending 'scikit-image, nsbatwm'. These are two slightly 
+different implementations of a _gaussian blur_, one from 
+`napari-simpleitk-image-processing` and the other from 
+`napari-segment-blobs-and-things-with-membranes`. Both should work perfectly 
+well for our image, but let's use the one ending with 'scikit-image, nsbatwm' 
+for now.
+
+If you click on this option, you should see a new panel appear on the right side 
+of Napari:
+
+![](fig/gaussian-options.png){alt="Screenshot of settings for gaussian blur in 
+Napari" width='40%'}
+
+Make sure you have 'nuclei(data)' selected on the 'image' row, then click run:
+
+![](fig/nuclei-blurred-1.png){alt="Nuclei image after gaussian blur with sigma 
+of 1" width='40%'}
+
+You should see a new image appear in the layer list called 'Result of 
+gaussian_blur', which is a slightly blurred version of the original nuclei image.
+
+Try increasing the 'sigma' value to three and clicking run again:
+
+![](fig/nuclei-blurred-3.png){alt="Nuclei image after gaussian blur with sigma 
+of 3" width='40%'}
+
+You should see the nuclei become even more blurred in the new image.
+
+What's happening here? What exactly does a gaussian blur do? A gaussian blur is 
+an example of a 'linear filter' which is used to manipulate pixel values in 
+images. When a filter is applied to an image, each pixel value is replaced by 
+some combination of the pixel values around it. For example, below is shown a 
+small, zoomed-in area of the nucleus image. Here, the value of the red pixel 
+could be affected by all values within the displayed 3x3 box:
+
+![](fig/nuclei-kernel-area.png){alt="Small zoomed-in area of the nucleus image 
+with a pixel highlighted in red. Around this pixel is shown a 
+3x3 box." width="60%"}
+
+Exactly what this effect is, and the size of the region it involves, is 
+controlled by the filter's 'kernel'. The kernel can be thought of as another 
+very small image, where the pixel values control the effect of the filter. An 
+example 3x3 kernel is shown below:
+
+![](fig/nuclei-kernel.png){alt="Left - small area of the nucleus image with a 
+pixel highlighted in red. Around this pixel is shown a 3x3 box. Right - example 
+of a 3x3 kernel"}
+
+Bear in mind that kernels can have many different sizes! For example 3x3, 5x5, 
+9x9... To determine the new value of a pixel (like the red pixel above), the 
```

Comment:
I changed it to: 
While the above image uses a 3x3 kernel, they can have many different sizes! Kernels tend to be odd in size: 3x3, 5x5, 9x9, etc. This is to allow the current pixel that you will be replacing to lie perfectly at the centre of the kernel, with equal spacing on all sides.

I thought it would be good to avoid 'operation is symmetric', as it's possible to have an odd kernel that produces an asymmetric effect, by loading higher filter values on one side of the kernel than the other.

Reply:
Of course, that's a good point! Yes I like the changes you have here instead of symmetric.